src/python/WMComponent/ArchiveDataReporter/ArchiveDataPoller.py

```diff
@@ -31,30 +32,31 @@ def setup(self, parameters):
          
         self.fwjrAPI = FWJRDBAPI(baseURL, dbname)
         self.wmarchiver = WMArchiver(self.config.ArchiveDataReporter.WMArchiverURL)
+        self.numDocsRetrievePerPolling = getattr(self.config.ArchiveDataReporter, "numDocsRetrievePerPolling", 1000)
+        self.numDocsUploadPerCall = getattr(self.config.ArchiveDataReporter, "numDocsUploadPerCall", 200)
         
-
     def algorithm(self, parameters):
         """
         get information from wmbs, workqueue and local couch
         """
         try:
             logging.info("Getting not archived data info from FWRJ db...")
-            data = self.fwjrAPI.getFWJRByArchiveStatus('ready')['rows']
+            data = self.fwjrAPI.getFWJRByArchiveStatus('ready', limit=1000)['rows']
             
-            #TODO need to send bulk update update bulk archive
-            jobIDs = []
-            archiverDocs = []
-            for job in data:
-                doc = createArchiverDoc(job)
-                archiverDocs.append(doc)
-                jobIDs.append(job["id"])
-                
-            response = self.wmarchiver.archiveData(archiverDocs)
+            for slicedData in grouper(data, self.numDocsUploadPerCall):
+                jobIDs = []
+                archiverDocs = []
+                for job in slicedData:
+                    doc = createArchiverDoc(job)
+                    archiverDocs.append(doc)
+                    jobIDs.append(job["id"])
+                    
+                response = self.wmarchiver.archiveData(archiverDocs)
             
-            # Partial success is not allowed either all the insert is successful of none is successful.
-            if response[0]['status'] == "ok" and len(response[0]['ids']) == len(jobIDs):
-                for docID in jobIDs:
-                    self.fwjrAPI.updateArchiveUploadedStatus(docID)
+                # Partial success is not allowed either all the insert is successful of none is successful.
+                if response[0]['status'] == "ok" and len(response[0]['ids']) == len(jobIDs):
+                    for docID in jobIDs:
+                        self.fwjrAPI.updateArchiveUploadedStatus(docID)
```

Comment:
@ticoann can you add the following line inside this if and outside the for please?
`logging.info("...successfully uploaded %d docs", len(jobIDs))`

It'll be helpful to see whether the component is doing the right thing.